src/basic/log.c

```diff
@@ -1689,6 +1690,44 @@ int log_syntax_invalid_utf8_internal(
                                    "String is not UTF-8 clean, ignoring assignment: %s", strna(p));
 }
 
+int log_syntax_parse_error_internal(
+                const char *unit,
+                const char *config_file,
+                unsigned config_line,
+                int error,
+                bool critical,
+                const char *file,
+                int line,
+                const char *func,
+                const char *lvalue,
+                const char *rvalue) {
+
+        PROTECT_ERRNO;
+        _cleanup_free_ char *escaped = NULL;
+
+        /* OOM is always critical. */
+        if (ERRNO_VALUE(error) == ENOMEM)
+                critical = true;
```

Comment:
Hmm, just return log_oom() here? I don't think log_syntax is really useful here, and we shouldn't try to do further memory allocation (utf8_escape_invalid)